File: app/src/main/java/com/example/focusstartsecondpart/features/guests/domain/GuestsInteractor.java
package com.example.focusstartsecondpart.features.guests.domain;

import com.example.focusstartsecondpart.features.guests.domain.model.Guest;
import com.example.focusstartsecondpart.features.guests.domain.model.Result;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Single;

public interface GuestsInteractor {
    Single<List<Guest>> loadGuests(int id);

    ArrayList<String> loadProfile(Guest guest);

    Single<Result> updateGuest(int eventId, Guest guest1);
}

File: app/src/main/java/com/example/focusstartsecondpart/features/guests/domain/GuestsInteractor.kt
package com.example.focusstartsecondpart.features.guests.domain

import com.example.focusstartsecondpart.features.guests.domain.model.Guest
import com.example.focusstartsecondpart.features.guests.domain.model.Result
import io.reactivex.Single
import java.util.*

interface GuestsInteractor {
    fun loadGuests(id: Int): Single<List<Guest>>

    fun loadProfile(guest: Guest): ArrayList<String>

    fun updateGuest(eventId: Int, guest1: Guest): Single<Result>
}
